### PR TITLE
fix(bucket): copy bucket id to clipboard

### PR DESCRIPTION
**Pull request by htuna07**
https://dreamharvesters.atlassian.net/browse/SP-144

### CODE REVIEW

**Comment by emre2345** [comment]
@htuna07 Is it possible writing a test for this?

File: stacks/spica/packages/material/clipboard/clipboard.directive.ts [hunk 1]
```diff
@@ -8,19 +8,31 @@ export class MatClipboardDirective {
...
+
+    document.body.removeChild(this.clipBoardElement);
+
     this.icon = "check";
     setTimeout(() => {
       this.icon = "info";
     }, 1000);
   }
+
+  prepareClipBoard(text: string) {
```
Comment (thesayyn) [inline stacks/spica/packages/material/clipboard/clipboard.directive.ts]: From the functional programming perspective, this method can return the newly created element.

Comment (thesayyn) [reply stacks/spica/packages/material/clipboard/clipboard.directive.ts]: Also, if we return the created element, we can eliminate `clipBoardElement` property. Much cleaner IMHO.

File: stacks/spica/packages/material/clipboard/clipboard.directive.ts [hunk 2]
```diff
@@ -8,19 +8,31 @@ export class MatClipboardDirective {
...
   copy() {
     if (!this.text) {
       return;
     }
-    const clipboard = document.createElement("input");
-    clipboard.innerText = this.text;
-    clipboard.select();
-    clipboard.setSelectionRange(0, this.text.length);
+
+    this.prepareClipBoard(this.text);
```
Comment (thesayyn) [inline stacks/spica/packages/material/clipboard/clipboard.directive.ts]: This line can be refactored to `const element = this.prepareClipBoard(this.text);` if the `prepareClipBoard` method can return the newly created  element.

**Comment by thesayyn** [comment]
NOTE: I strongly recommend you to do the changes that I have described above.

File: stacks/spica/packages/material/clipboard/clipboard.directive.ts [hunk 3]
```diff
@@ -13,14 +13,29 @@ export class MatClipboardDirective {
...
-    clipboard.innerText = this.text;
-    clipboard.select();
-    clipboard.setSelectionRange(0, this.text.length);
-    document.execCommand("copy");
+
+    const element = this.prepareElement(this.text);
+
+    this.copyToClipBoard(element);
+
+    document.body.removeChild(element);
```
[the remark this answers is not in the document]
Comment (htuna07) [reply stacks/spica/packages/material/clipboard/clipboard.directive.ts]: if it moves into **copyToClipBoard** method, the copied value won't be testable, cause the element will be removed from the body.

[the remark this answers is not in the document]
Comment (thesayyn) [reply stacks/spica/packages/material/clipboard/clipboard.directive.ts]: Okay, let's leave a TODO on top of this method to reconsider rewrite of this directive using Clipboard API once it is available in evergreen browsers.

[the remark this answers is not in the document]
Comment (thesayyn) [reply stacks/spica/packages/material/clipboard/clipboard.directive.ts]: https://caniuse.com/#search=Clipboard%20API

[the remark this answers is not in the document]
Comment (thesayyn) [reply stacks/spica/packages/material/clipboard/clipboard.directive.ts]: Also, keep in mind that these tests are not confirming that this directive will work in any browser.

**Pull request merged**